controller/service.go

```diff
@@ -59,11 +60,16 @@ func (c *controller) convergeBalancer(l log.Logger, key string, svc *v1.Service)
 	// The assigned LB IP(s) is the end state of convergence. If there's
 	// none or a malformed one, nuke all controlled state so that we
 	// start converging from a clean slate.
+	// Store hostnames for further processing.
 	for i := range svc.Status.LoadBalancer.Ingress {
 		ip := svc.Status.LoadBalancer.Ingress[i].IP
+		hostname := svc.Status.LoadBalancer.Ingress[i].Hostname
```

Comment:
not related to this change, but if we are using i only to access the array, we can convert this to _, ingress and access the element directly.